framework/src/wifi_manager/wifi_net.c

```diff
@@ -0,0 +1,220 @@
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <sys/socket.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <time.h>
+#include <sched.h>
+#include <errno.h>
+#include <debug.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include "wifi_utils.h"
+#include "wifi_net.h"
+
+#define ptr2uint32(ptr) (uint32_t)ptr[3] |     \
+                ((uint32_t)ptr[2] << 8)  |     \
+                ((uint32_t)ptr[1] << 16) |     \
+                ((uint32_t)ptr[0] << 24)
+
+#define NTP2UNIX_TRANLSLATION 2208988800u
+#define NTP_VERSION           3
+static void convert_ntp_time(FAR uint8_t *timestamp, unsigned int *sec, unsigned int *usec)
+{
+	time_t seconds;
+	uint32_t frac;
+	uint32_t nsec;
+#ifdef CONFIG_HAVE_LONG_LONG
+	uint64_t tmp;
+#else
+	uint32_t a16;
+	uint32_t b0;
+	uint32_t t32;
+	uint32_t t16;
+	uint32_t t0;
+#endif
+
+	/* NTP timestamps are represented as a 64-bit fixed-point number, in
+	 * seconds relative to 0000 UT on 1 January 1900.  The integer part is
+	 * in the first 32 bits and the fraction part in the last 32 bits, as
+	 * shown in the following diagram.
+	 *
+	 *    0                   1                   2                   3
+	 *    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+	 *   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 *   |                         Integer Part                          |
+	 *   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 *   |                         Fraction Part                         |
+	 *   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 */
+
+	seconds = ptr2uint32(timestamp);
+
+	/* Translate seconds to account for the difference in the origin time */
+
+	if (seconds > NTP2UNIX_TRANLSLATION) {
+		seconds -= NTP2UNIX_TRANLSLATION;
+	}
+
+	/* Conversion of the fractional part to nanoseconds:
+	 *
+	 *  NSec = (f * 1,000,000,000) / 4,294,967,296
+	 *       = (f * (5**9 * 2**9) / (2**32)
+	 *       = (f * 5**9) / (2**23)
+	 *       = (f * 1,953,125) / 8,388,608
+	 */
+
+	frac = ptr2uint32((timestamp + 4));
+#ifdef CONFIG_HAVE_LONG_LONG
+	/* if we have 64-bit long long values, then the computation is easy */
+
+	tmp = ((uint64_t) frac * 1953125) >> 23;
```

Comment:
same as previous
```
(uint64_t) frac => (uint64_t)frac
```

Reply:
formatter doesn't fix this problem. I mannually fix it.